hail/python/test/hail/methods/test_pca.py

```diff
@@ -166,48 +162,156 @@ def spectra_helper(spec_func):
         np.testing.assert_allclose(norm_of_diff, spec_func(k + 1, k), rtol=1e-02, err_msg=f"Norm test failed on triplet {triplet} ")
         np.testing.assert_allclose(singulars, np.diag(sigma)[:k], rtol=1e-01, err_msg=f"Failed on triplet {triplet}")
 
+
+def spec1(j, k):
+    return 1/j
+
+
+def spec2(j, k):
+    if j == 1:
+        return 1
+    if j <= k:
+        return 2 * 10**-5
+    else:
+        return (10**-5) * (k + 1)/j
+
+
+def spec3(j, k):
+    if j <= k:
+        return 10**(-5*(j-1)/(k-1))
+    else:
+        return (10**-5)*(k+1)/j
+
+
+def spec4(j, k):
+    if j <= k:
+        return 10**(-5*(j-1)/(k-1))
+    elif j == (k + 1):
+        return 10**-5
+    else:
+        return 0
+
+
+def spec5(j, k):
+    if j <= k:
+        return 10**-5 + (1 - 10**-5)*(k - j)/(k - 1)
+    else:
+        return 10**-5 * math.sqrt((k + 1)/j)
+
+
 @fails_service_backend(reason='persist_ir')
 def test_spectra_1():
-    def spec1(j, k):
-        return 1/j
     spectra_helper(spec1)
 
+
 @fails_service_backend(reason='persist_ir')
 def test_spectra_2():
-    def spec2(j, k):
-        if j == 1:
-            return 1
-        if j <= k:
-            return 2 * 10**-5
-        else:
-            return (10**-5) * (k + 1)/j
     spectra_helper(spec2)
 
+
 @fails_service_backend(reason='persist_ir')
 def test_spectra_3():
-    def spec3(j, k):
-        if j <= k:
-            return 10**(-5*(j-1)/(k-1))
-        else:
-            return (10**-5)*(k+1)/j
     spectra_helper(spec3)
 
+
 @fails_service_backend(reason='persist_ir')
 def test_spectra_4():
-    def spec4(j, k):
-        if j <= k:
-            return 10**(-5*(j-1)/(k-1))
-        elif j == (k + 1):
-            return 10**-5
-        else:
-            return 0
     spectra_helper(spec4)
 
+
 @fails_service_backend(reason='persist_ir')
 def test_spectra_5():
-    def spec5(j, k):
-        if j <= k:
-            return 10**-5 + (1 - 10**-5)*(k - j)/(k - 1)
-        else:
-            return 10**-5 * math.sqrt((k + 1)/j)
     spectra_helper(spec5)
+
+
+@fails_service_backend(reason='persist_ir')
+def spectral_moments_helper(spec_func):
+    for triplet in [(20, 1000, 1000)]:
+        k, m, n = triplet
+        min_dim = min(m, n)
+        sigma = np.diag([spec_func(i+1, k) for i in range(min_dim)])
+        seed = 1025
+        np.random.seed(seed)
+        U = np.linalg.qr(np.random.normal(0, 1, (m, min_dim)))[0]
+        V = np.linalg.qr(np.random.normal(0, 1, (n, min_dim)))[0]
+        A = U @ sigma @ V.T
+        mt_A = matrix_table_from_numpy(A)
+
+        moments, stdevs = hl._spectral_moments(_make_tsm(mt_A.ent, 128), 7)
```

Comment:
Should we be checking the stddevs too?

Reply:
I'm not sure how to test them (they're just an estimate for how accurate the moment estimates are), and from hand testing it looks like they might be off anyways. I alerted Alex and Patrick C., so I'll wait for feedback from them on how the error estimates are working.

Reply:
Are we still waiting for feedback from them, or should I go ahead and do next round of review?

Reply:
Sorry, I meant based on feedback I might need to adjust how stdevs are computed, but I don't think there's any need to hold up this PR for that. This is all experimental, and will only have one user for a while.